### PR TITLE
feat: add "good first issue" label to all GitHub repos

### LINKED ISSUE #51: [GH Request] The openedx org should have a "good-first-issue" label
### Firm Name

Overhang.IO

### Urgency

Low urgency

### Requested Change

Add a "good-first-issue" issue label to the openedx organisation repository defaults. This can be done by an admin here: https://github.com/organizations/openedx/settings/repository-defaults 

This issue was initially discussed here: https://github.com/openedx/tutor-contrib-multi/issues/12#issuecomment-1412372920

### Reasoning

New contributors who are looking for issues to help the project need a way to identify issues that are not too complex or don't require too much internal knowledge. It is customary to tag such issues with a "good first issue" label.

Here are two repos that use this label:
https://github.com/overhangio/tutor/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+
https://github.com/openedx/wg-developer-experience/issues?q=is%3Aopen+is%3Aissue+label%3Agood-first-issue

We should add this label to the openedx organisation such that any newly-created repository automatically has it by default.

The label can be dash-separated ("good-first-issue") or space-separated ("good first issue"). I don't have a strong preference, but I would lean toward dash-separated, because it's easier to type in a URL.

### PR DESCRIPTION
**Pull request by regisb**
The label will be added to all existing repos. Note that this might conflict with the recent "good first issue :tada:" default label that is part of the openedx org. That "tada" label will probably have to be renamed, and repos that already have it should migrate the issues to use the new label.

Color is a sweet green. Alternative suggestions are welcome.

Close #51.

### CODE REVIEW

**Comment by kdmccormick** [comment]
I am in support of this 👍🏻 

However, I would want to test it it out to make sure it converts other spelling of the label as we meant it to (https://github.com/openedx/terraform-github/issues/51#issuecomment-1428265498).

I think we need to decide whether we want emoji in the labels or not. I think the emoji are fun and eye-catching, but GitHub comes with `good first issue` as a default and maybe there's some sense in just sticking to that. I'd want to be consistent across all of our standard labels.

Finally, @sarina @feanil do you think we should just get rid of the org-wide default labels from our GitHub settings? It just seems like something that can get out-of-sync with this list and cause confusion. Alternatively, we could do an API request to pull the org-wide default labels from GitHub settings, using them as the source for this script.

**Comment by regisb** [comment]
> However, I would want to test it it out to make sure it converts other spelling of the label as we meant it to (https://github.com/openedx/terraform-github/issues/51#issuecomment-1428265498).

I can confirm that: `_simplify_label("good first issue :tada:")` returns `"goodfirstissue". Same for `_simplify_label("good-first-issue")`, `_simplify_label("good first issue")`, etc. So we should be fine here.

I don't have a particular opinion on the choice of the label and color, so I'll leave the decision to whom it may concern.

**Comment by feanil** [comment]
@regisb is there a reason not to converge on the repository default that includes `:tada` ?

I think if we go with this, in the short term we can update the repository default manually to match this.  In the longterm, I think we should have this check also update the repository defaults that match.

**Comment by kdmccormick** [comment]
My only hang-up with the `:tada` was that plain-old `good first issue` comes out-of-the-box with a new repo in a new org, and so using it might boost openedx's "SEO" for devs looking for a project that they can contribute to. This is entirely theoretical, and it looks like [GitHub uses ML to curate good first issues anyway](https://github.blog/2020-01-22-how-we-built-good-first-issues/), so I don't think it's worth worrying about.

tl;dr -- If nobody objects, let's include the 🎉

**Comment by regisb** [comment]
Updated the lable, please review again :)

**Review by sarina — APPROVED** [review]
Works for me. Did anyone else want to weigh in?

I think once we merge @feanil may need to kick off a manual sync, unless we've completed automation work.

**Review by feanil — APPROVED** [review]

**Comment by feanil** [comment]
Correct, the update is still manual but I'll run it now.

**Comment by openedx-webhooks** [comment]
@regisb 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/terraform-github&entry.867055334=https://github.com/openedx/terraform-github/pull/56&entry.1484655318=https://github.com/regisb&entry.752974735=2023-02-13+16:39&entry.1917517419=2023-02-27+16:39&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by feanil** [comment]
Run everywhere, this should be all set.
